feature/gnoi/packet_link_qualification/tests/packet_link_qualification_test/packet_link_qualification_test.go

```diff
@@ -422,11 +439,16 @@ func TestLinkQualification(t *testing.T) {
 	// The packet counters between Generator and Reflector mismatch tolerance level in percentage
 	var tolerance float64 = 0.0001
 
-	if ((math.Abs(float64(generatorPktsSent)-float64(reflectorPktsRxed)))/(float64(generatorPktsSent)+float64(reflectorPktsRxed)+tolerance))*200.00 > tolerance {
-		t.Errorf("The difference between packets received count at Reflector and packets sent count at Generator is greater than %0.4f percent: generatorPktsSent %v, reflectorPktsRxed %v", tolerance, generatorPktsSent, reflectorPktsRxed)
-	}
-	if ((math.Abs(float64(reflectorPktsSent)-float64(generatorPktsRxed)))/(float64(reflectorPktsSent)+float64(generatorPktsRxed)+tolerance))*200.00 > tolerance {
-		t.Errorf("The difference between packets received count at Generator and packets sent count at Reflector is greater than %0.4f percent: reflectorPktsSent %v, generatorPktsRxed %v", tolerance, reflectorPktsSent, generatorPktsRxed)
+	if deviations.PLQReflectorStatsUnsupported(dut1) {
+		if (math.Abs(float64(generatorPktsSent)-float64(generatorPktsRxed))/float64(generatorPktsSent))*100.00 > tolerance {
```

Comment:
Why the change in math for the generator?

Based on the description, only the reflector check should be under the deviation.

Reply:
sorry i didn't understand , the existing checks are not changed they are moved to the else condition.
The existing checks are wrt to Generator sent - Reflector rcvd & Reflector sent - Generator rcvd,
so this wont work if reflector stats are unsupported, hence generator specific check is added under the deviation.

Reply:
Ah, I missed that part, sorry. Got it, thx.